server/src/test/java/io/crate/execution/ddl/tables/AddColumnTaskTest.java

```diff
@@ -397,4 +399,47 @@ public void test_collect_new_columns() throws Exception {
             );
         }
     }
+
+    @Test
+    public void test_table_version_less_than_5_5_oid_is_not_assigned() throws Exception {
+        var e = SQLExecutor.builder(clusterService)
+            .addTable(
+                "create table tbl (x int)",
+                Settings.builder().put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.V_5_4_0).build()
+            )
+            .build();
+
+        DocTableInfo tbl = e.resolveTableInfo("tbl");
+        try (IndexEnv indexEnv = new IndexEnv(
+            THREAD_POOL,
+            tbl,
+            clusterService.state(),
+            Version.V_5_4_0,
+            createTempDir()
+        )) {
+            var addColumnTask = new AddColumnTask(e.nodeCtx, imd -> indexEnv.mapperService());
+
+            SimpleReference colToAdd = new SimpleReference(
+                new ReferenceIdent(tbl.ident(), "y"),
+                RowGranularity.DOC,
+                DataTypes.STRING,
+                2,
+                null
+            );
+
+            List<Reference> columns = List.of(colToAdd);
+            var request = new AddColumnRequest(
+                tbl.ident(),
+                columns,
+                Map.of(),
+                new IntArrayList()
+            );
+            ClusterState newState = addColumnTask.execute(clusterService.state(), request);
+            DocTableInfo newTable = new DocTableInfoFactory(e.nodeCtx).create(tbl.ident(), newState);
+
+            Reference addedColumn = newTable.getReference(colToAdd.column());
+            // Used to be 2, ie OID used to be assigned on 5.4 tables.
```

Comment:
The comment is a bit confusing, why `used to be`?
you mean for >= 5.5 it should be a value `> 0`?

Reply:
I agree that this comment is confusing. As the test name states already what it is about, I'd remove the comment. Otherwise I vote for @matriv's suggestion.

Reply:
I removed the comment, test name is sufficient I think